src/main/java/com/denizenscript/depenizen/bukkit/clientizen/network/DataSerializer.java

```diff
@@ -0,0 +1,111 @@
+package com.denizenscript.depenizen.bukkit.clientizen.network;
+
+import com.denizenscript.denizencore.utilities.debugging.Debug;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutput;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+public class DataSerializer {
+
+    private final DataOutput output;
+    private final ByteArrayOutputStream outputStream;
+
+    public DataSerializer() {
+        outputStream = new ByteArrayOutputStream();
+        output = new DataOutputStream(outputStream);
+    }
+
+    public DataSerializer writeBoolean(boolean bool) {
+        try {
+            output.writeBoolean(bool);
+        }
+        catch (IOException e) {
+            Debug.echoError(new IllegalStateException(e));
+        }
+        return this;
+    }
+
+    public DataSerializer writeInt(int i) {
+        try {
+            output.writeInt(i);
+        }
+        catch (IOException e) {
```

Comment:
Why would an IO Exception ever occur? 0.o

Reply:
<img width="293" alt="image" src="https://github.com/DenizenScript/Depenizen/assets/31237389/42496f2f-96c6-4a7f-a2e8-f14b9caef019">

We could maybe switch to using Netty `ByteBuf`s? they're more specifically aimed at network stuff, and would let us have matching code with the client-side equivalent which might be a nice benefit

Reply:
An error occurs if an error occurs. Wow, incredible.

Also if you use ByteBuf you can unify the code with Bungee's packet could, which would be nice too.